core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1696,45 +1696,37 @@ class KafkaApis(val requestChannel: RequestChannel,
   def handleSyncGroupRequest(request: RequestChannel.Request, requestLocal: RequestLocal): Unit = {
     val syncGroupRequest = request.body[SyncGroupRequest]
 
-    def sendResponseCallback(syncGroupResult: SyncGroupResult): Unit = {
-      requestHelper.sendResponseMaybeThrottle(request, requestThrottleMs =>
-        new SyncGroupResponse(
-          new SyncGroupResponseData()
-            .setErrorCode(syncGroupResult.error.code)
-            .setProtocolType(syncGroupResult.protocolType.orNull)
-            .setProtocolName(syncGroupResult.protocolName.orNull)
-            .setAssignment(syncGroupResult.memberAssignment)
-            .setThrottleTimeMs(requestThrottleMs)
-        ))
+    def sendResponse(response: AbstractResponse): Unit = {
+      trace("Sending sync group response %s for correlation id %d to client %s."
+        .format(response, request.header.correlationId, request.header.clientId))
+      requestHelper.sendResponseMaybeThrottle(request, requestThrottleMs => {
+        response.maybeSetThrottleTimeMs(requestThrottleMs)
+        response
+      })
```

Comment:
thoughts on reusing this? it's identical to handleJoinGroupRequest except for the response type and it looks like we can use this for the other coordinator APIs as well

Reply:
I have mainly kept `sendResponse` because of the `trace` log there. I have refactored this a but in the last commit. Let me know what you think.

Reply:
thanks, makes sense to me.